scripts/googledrive/connect_external_accounts.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+First, do pre-merge migration (found at https://github.com/CenterForOpenScience/osf.io/pull/4396 )
+
+Then, merge and change the user_settings field of GoogleDriveNodeSettings to foreign_user_settings
+"""
+import sys
+import logging
+from modularodm import Q
+
+from website.app import init_app
+from scripts import utils as script_utils
+from framework.transactions.context import TokuTransaction
+
+from website.addons.googledrive.model import GoogleDriveNodeSettings
+
+logger = logging.getLogger(__name__)
+
+
+def do_migration():
+    for node_addon in GoogleDriveNodeSettings.find(Q('foreign_user_settings', 'ne', None)):
+        f_id = None
+        user_addon = node_addon.foreign_user_settings
+        user = user_addon.owner
```

Comment:
As discussed in person, the `user_addon` with `node_addon.foreign_user_settings == None` was created with the new OAUTH refactor and did not get deleted from the db. When testing with a new db, the migration worked.